src/features/dashboard/components/ApiTokenTable/__tests__/index.test.tsx

```diff
@@ -24,110 +25,98 @@ mockUseDeleteToken.mockImplementation(() => ({
   deleteToken: mockDeleteToken,
 }));
 
-describe('Api Token Table', () => {
-  afterEach(() => {
-    cleanup();
-    jest.clearAllMocks();
-  });
-
-  it('Should render loading when isLoadingTokens is truthy ', async () => {
-    mockUseApiToken.mockImplementationOnce(() => ({
-      tokens: [],
-      isLoadingTokens: true,
-    }));
+jest.mock('@site/src/hooks/useDeviceType');
+const mockDeviceType = useDeviceType as jest.MockedFunction<
+  () => Partial<ReturnType<typeof useDeviceType>>
+>;
+mockDeviceType.mockImplementation(() => ({
+  deviceType: 'desktop',
+}));
 
-    render(<ApiTokenTable />);
+const fakeTokens: TTokensArrayType = [
+  {
+    display_name: 'This is my first token',
+    last_used: '',
+    scopes: ['read', 'trade'],
+    token: 'first_token',
+    valid_for_ip: '',
+  },
+];
+
+beforeEach(() => {
+  mockUseApiToken.mockImplementation(() => ({
+    tokens: fakeTokens,
+    isLoadingTokens: true,
+  }));
+  render(<ApiTokenTable />);
+});
 
-    const loadingElement = await screen.findByTestId('dt_spinner');
-    expect(loadingElement).toBeVisible();
-  });
+afterEach(() => {
+  jest.clearAllMocks();
 });
```

Comment:
Before each and after each should be inside the describe callback

Reply:
Yes, for these tests I've understood the syntax and a bit of the basics. Will keep practicing these to better write these tests.